src/components/ErrorPage.tsx

```diff
@@ -19,11 +19,15 @@ export const ErrorPage = () => {
         <h1 className="text-4xl mb-8 font-bold" data-testid="unexpected-error">
           An unexpected error occurred
         </h1>
-        {isRouteErrorResponse(error) && (
-          <p className="mb-8">
-            {error.status}: {error.data}
-          </p>
-        )}
+        <p className="mb-8">
+          {isRouteErrorResponse(error) ? (
+            <>
+              {error.status}: {error.data}
+            </>
+          ) : (
+            <>JSON.stringify(error)</>
```

Comment:
This is an HTML string, this isn't an evaluated JS expression. 

We should do something like [this](https://github.com/remix-run/react-router/discussions/9628#discussioncomment-5555901) that I found from [here](https://stackoverflow.com/questions/75944820/whats-the-correct-type-for-error-in-userouteerror-from-react-router-dom).

We need to type narrow `error : unknown` to get a string message from it. The code `JSON.stringify(error)` does not work as a blanket case. 

https://github.com/user-attachments/assets/36756f1b-42cb-41d8-a98e-acb55d416d0f

Reply:
lmao thank you for this, wow this might be the worst PR I've ever put up